java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspTemplateUI.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.lsp.server.protocol;
+
+import com.google.gson.JsonPrimitive;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.eclipse.lsp4j.WorkspaceFolder;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.loaders.DataFolder;
+import org.openide.loaders.DataObject;
+import org.openide.loaders.DataObjectNotFoundException;
+import org.openide.nodes.Node;
+import org.openide.util.NbBundle;
+
+class LspTemplateUI {
+    private LspTemplateUI() {
+    }
+
+    static CompletableFuture<Object> createFromTemplate(String templates, NbCodeLanguageClient client, ExecuteCommandParams params) {
+        final FileObject fo = FileUtil.getConfigFile(templates);
+        final DataFolder folder = DataFolder.findFolder(fo);
+        return displayUI(folder, client, params);
```

Comment:
What about returning `null` (no actuon) or `IOException` if the `fo` is missing rather than `IllegalArgumentException` with user-cryptic message ?

Reply:
I think I want to return `CompletableFuture` as that's what the LSP then expects. Anyway 3b8dec3b46b4 shall eliminate (explicit) use of `IllegalArgumentException` and return return `null` from the future when user cancels one of the quick picks.

Reply:
Sure I meant `completedFuture(null)`, just not IAE. Sorry.